rtengine/dcraw.cc

```diff
@@ -6439,7 +6439,8 @@ int CLASS parse_tiff_ifd (int base)
 	break;
       case 258:				/* BitsPerSample */
       case 61443:
-	tiff_ifd[ifd].samples = len & 7;
+	if (tiff_ifd[ifd].samples != 4)
+	    tiff_ifd[ifd].samples = len & 7; // possible bug? should this be .bps instead of .samples ?
```

Comment:
FYI LibRaw avoids resetting the samples if the tag is 258 and it is already set.
https://github.com/LibRaw/LibRaw/blob/6fffd414bfda63dfef2276ae07f7ca36660b8888/src/metadata/tiff.cpp#L438-L439

Reply:
This will change the samples if it's anything other than 4, whereas LibRaw only does so if samples hasn't been set. I wonder if this will be problematic.